### PR TITLE
Fix applying parameters to Cloudflare API on GET requests

### PR DESCRIPTION
**Pull request by qip**
For instance, the per-page parameter wasn't passed to Cloudflare API when retrieving DNS records.
Also, it works better when filtering on server-side since pagination hasn't implemented.

### CODE REVIEW

**Comment by NewFuture** [comment]
cloudflare.py 能否单独checkout 一个分支PR，方便merge

**Comment by NewFuture** [comment]
@tongyifan 有空可以review一下

**Comment by tongyifan** [comment]
@NewFuture 好，起床就看

File: dns/cloudflare.py [hunk 1]
```diff
@@ -46,6 +46,8 @@ def request(method, action, param=None, **params):
         params = json.dumps(params)
     else:
         params = urllib.urlencode(params)
+    if method == 'GET':
+        action = action + '?' + params
     conn.request(method, '/client/v4/zones' + action, params,
```
Comment (tongyifan) [inline dns/cloudflare.py:51]: Yep that's a bug. I misunderstood third parameter in [HTTPSConnection.request](https://docs.python.org/3/library/http.client.html#http.client.HTTPConnection.request) . Good job!

**Comment by qip** [comment]
See #38

**Pull request closed without merging**